### DISCUSSION TITLE
Why: char *label vs const char *label

### DISCUSSION THREAD

**Discussion by marcelstoer** [Bodmer/TFT_eSPI#2552] [General]
Your element constructors use `char *label` rather than `const char *label` e.g. https://github.com/Bodmer/TFT_eSPI/blob/master/Extensions/Button.h#L16. Why is that so?

Hence, instead of passing a string literal created inline/on-the-fly I need to create a `char[]`.

```c++
myButton.initButton(&tft, 400, 30, 141, 43, 0xFFFF, 0x2A54, 0xFFFF, "Foo", 2);
```
--> the compiler complains "warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]"

To quiet it, I need to use something like this.

```c++
char label[] = "Foo";
myButton.initButton(&tft, 400, 30, 141, 43, 0xFFFF, 0x2A54, 0xFFFF, label, 2);
```

**Comment by Bodmer** [comment]
Yes, there are a few cases where this warning appears, so I need to update these instances at some point.

**Reply by marcelstoer** [reply]
Ok, I found four such cases but only two in the API: #2571 <- PR landed on `dev`
